client/components/signup-site-preview/iframe-utils.ts

```diff
@@ -0,0 +1,127 @@
+// This module is for dealing with DOM elements from inside the the site preview
+// iframe. This means that not every modern JS feature will be available to use
+// because they won't have been polyfilled e.g. a NodeList from the iframe isn't
+// iterable using a for-of loop.
+/* eslint no-for-of-loops/no-for-of-loops: 2 */
```

Comment:
My worry is that there needs to be an awareness of this issue in the first place to avoid future bugs.
If somebody else were to replicate the same problem they would have had to add `* eslint no-for-of-loops/no-for-of-loops: 2 */` to their file for the rule to be of any use at all. The fact that it crossed their mind by their own knowledge or experience means they were likely not going to be adding problematic code in the first place.
Really, with linting, we want it to serve as a way to catch issues that the dev isn't already aware of and this solution doesn't seem to do that. Also, as a side effect, it looks like we can't use for-of loops in this whole file now without fiddling with extra 'off...on' eslint comments.